src/glomap/estimators/bundle_adjustment.cc

```diff
@@ -113,119 +112,118 @@ void BundleAdjuster::Reset() {
 }
 
 void BundleAdjuster::AddPointToCameraConstraints(
-    std::unordered_map<rig_t, Rig>& rigs,
-    std::unordered_map<camera_t, colmap::Camera>& cameras,
-    std::unordered_map<frame_t, Frame>& frames,
-    std::unordered_map<image_t, Image>& images,
-    std::unordered_map<point3D_t, Point3D>& tracks) {
-  for (auto& [track_id, track] : tracks) {
+    colmap::Reconstruction& reconstruction) {
+  for (const auto& [track_id, track] : reconstruction.Points3D()) {
```

Comment:
nit: We may want to make another pass in another PR later and rename "track" to "point3D" for consistency.